contrib/google.golang.org/grpc/option.go

```diff
@@ -60,24 +62,37 @@ func defaults(cfg *config) {
 }
 
 func clientDefaults(cfg *config) {
-	cfg.serviceName = namingschema.NewDefaultServiceName(
+	sn := namingschema.NewDefaultServiceName(
 		defaultClientServiceName,
 		namingschema.WithOverrideV0(defaultClientServiceName),
 	).GetName()
+	cfg.serviceName = func() string { return sn }
 	cfg.spanName = namingschema.NewGRPCClientOp().GetName()
 	defaults(cfg)
 }
 
 func serverDefaults(cfg *config) {
-	cfg.serviceName = namingschema.NewDefaultServiceName(defaultServerServiceName).GetName()
+	// We check for a configured service name, so we don't break users who are accidentally creating their server
+	// before the call `tracer.Start()`
+	if globalconfig.ServiceName() != "" {
```

Comment:
Did you intend to use the exported function from namingschema?
```suggestion
	if namingschema.HasConfiguredServiceName() != "" {
```

Reply:
Oops, actually my intent was to not add a new exported function at all (and I didn't realize until after I had already added so I forgot to remove it, will do that now)